File: 백준/Silver/9465. 스티커/스티커.java
    import java.util.*;
    import java.io.*;
    public class Main {

    	public static void main(String[] args) throws IOException {
    	BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
    	BufferedWriter bw=new BufferedWriter(new OutputStreamWriter(System.out));
    	int tc=Integer.parseInt(br.readLine());
    	for(int i=0;i<tc;i++) {
    		int num=Integer.parseInt(br.readLine());
    		int arr[][]=new int[2][num];
    		int dp[][]=new int [2][num];
    		String arr1[]=br.readLine().split(" ");
    		String arr2[]=br.readLine().split(" ");
    		for(int j=0;j<num;j++) {
    			arr[0][j]=Integer.parseInt(arr1[j]);
    			arr[1][j]=Integer.parseInt(arr2[j]);
    		}
    		for(int k=0;k<num;k++) {
    			if(k==0) {
    				dp[0][k]=arr[0][k];
    				dp[1][k]=arr[1][k];
    			}
    			else if(k==1) {
    				dp[0][k]=arr[0][k]+arr[1][k-1];
    				dp[1][k]=arr[1][k]+arr[0][k-1];
    				}
    			else {
    			dp[0][k]=Math.max(dp[1][k-1],dp[1][k-2])+arr[0][k];
    			dp[1][k]=Math.max(dp[0][k-1], dp[0][k-2])+arr[1][k];
    			}
    		}
    		System.out.println(Math.max(dp[0][num-1], dp[1][num-1]));
    		}
    	}
    }    

File: 백준/Silver/9465. 스티커/스티커.py
tc = int(input())
for i in range(tc):
    n = int(input())
    arr = [list(map(int,input().split())) for _ in range(2)]
    dp = [[0]*n for _ in range(2)]
    for i in range(n):
        if i == 0:
            dp[0][0],dp[1][0] = arr[0][0],arr[1][0]
        elif i == 1:
            dp[0][1],dp[1][1] = arr[1][0]+arr[0][1],arr[0][0]+arr[1][1]
        else:
            dp[0][i] = max(arr[0][i]+dp[1][i-1],arr[0][i]+dp[1][i-2])
            dp[1][i] += max(arr[1][i]+dp[0][i-1],arr[1][i]+dp[0][i-2])
    print(max(ans[n-1] for ans in dp))